### PR TITLE
Github getBranches does not return protected bit anymore

### PR DESCRIPTION
**Pull request by sebastienvas**
Doing 2 requests instead of one. Get all branches, and then get protected.

### CODE REVIEW

File: prow/github/client.go [hunk 1]
```diff
@@ -856,13 +857,36 @@ func (c *Client) GetRepos(org string, isUser bool) ([]Repo, error) {
...
-	return branches, err
+	}, &allBranches)
+	if err != nil {
+		return nil, err
+	}
+	// the list branch api does not set protected parameter on the response.
+	// Doing another call to only get protected repos
+	_, err = c.request(&request{
+		method:    http.MethodGet,
+		path:      fmt.Sprintf("/repos/%s/%s/branches?protected=true", org, repo),
```
Comment (fejta) [inline prow/github/client.go]: Please make this a boolean argument to `GetBranches()`

Move the logic to combine the two results into UpdateRepo() - https://github.com/kubernetes/test-infra/blob/master/prow/cmd/branchprotector/protect.go#L241

Comment (sebastienvas) [reply prow/github/client.go]: that s what the GetBranches interface defines. It is just that the github client fails to implement it.

Comment (cjwagner) [reply prow/github/client.go]: I agree with Erick, this logic doesn't belong in the Github client. What `GetBranches` interface are you referring to?

**Comment by fejta** [comment]
Possibly superseded by https://github.com/kubernetes/test-infra/pull/8085 (although I can pull out the GetBranches() change if this goes in)

File: prow/github/client.go (hunk 1 above)
Comment (sebastienvas) [reply prow/github/client.go]: https://github.com/kubernetes/test-infra/blob/master/prow/cmd/branchprotector/protect.go#L146. GetBranch expects a []Branch with protected set to false or true depending on protection status. I feel that updating the interface to match the behavior of the client is backward. It complicates the branch protection code for no reason. Note also that this is bug in github, that may be resolved at some point.

**Pull request closed without merging**